redbeams/src/main/java/com/sequenceiq/redbeams/sync/DBStackStatusSyncService.java

```diff
@@ -0,0 +1,94 @@
+package com.sequenceiq.redbeams.sync;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+
+import com.sequenceiq.cloudbreak.cloud.CloudConnector;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.context.CloudContext;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.cloud.model.CloudCredential;
+import com.sequenceiq.cloudbreak.cloud.model.ExternalDatabaseStatus;
+import com.sequenceiq.cloudbreak.cloud.model.Location;
+import com.sequenceiq.redbeams.api.model.common.DetailedDBStackStatus;
+import com.sequenceiq.redbeams.api.model.common.Status;
+import com.sequenceiq.redbeams.converter.cloud.CredentialToCloudCredentialConverter;
+import com.sequenceiq.redbeams.domain.stack.DBStack;
+import com.sequenceiq.redbeams.dto.Credential;
+import com.sequenceiq.redbeams.service.CredentialService;
+import com.sequenceiq.redbeams.service.stack.DBStackStatusUpdater;
+
+import static com.sequenceiq.cloudbreak.cloud.model.AvailabilityZone.availabilityZone;
+import static com.sequenceiq.cloudbreak.cloud.model.Location.location;
+import static com.sequenceiq.cloudbreak.cloud.model.Region.region;
+
+@Component
+public class DBStackStatusSyncService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DBStackStatusSyncService.class);
+
+    @Inject
+    private CredentialService credentialService;
+
+    @Inject
+    private CredentialToCloudCredentialConverter credentialConverter;
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private DBStackStatusUpdater dbStackStatusUpdater;
+
+    public void sync(DBStack dbStack) {
+        DetailedDBStackStatus detailedDBStackStatus = getDetailedDBStackStatusFromProvider(dbStack);
+        Status status = detailedDBStackStatus.getStatus();
```

Comment:
if any error is occurring at `External DB status lookup failed`, you will get an NPE

Reply:
Good catch